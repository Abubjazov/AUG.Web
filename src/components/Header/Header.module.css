.root {
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 60px;
  padding: 0 20px;
}

.wrapper {
  display: flex;
  align-items: center;
}

.logo-text {
  font-size: 14px;
  font-weight: 900;
  line-height: 16px;
  color: var(--logotext);
  text-align: left;
  text-transform: uppercase;
  animation: var(--animation-visible);
}

.red-dot {
  font-size: 20px;
  color: var(--red);
}

.burger {
  display: flex;
  align-items: center;
  width: 30px;
  height: 30px;
}
