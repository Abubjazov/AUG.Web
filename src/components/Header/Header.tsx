import { FC } from 'react'

import { useAppDispatch, useAppSelector } from 'store/hooks'
import { setMenuState, setDappletSettingsState } from 'store/slices/layoutSlice'
import ExtensionState from 'uikit/ExtensionState'
import HeaderSettings from 'uikit/HeaderSettings'
import SvgIcon from 'uikit/SvgIcon'

import styles from './Header.module.css'

interface HeaderProps {
  windowInnerWidth: number
}

const Header: FC<HeaderProps> = ({ windowInnerWidth }) => {
  const { menuOpened, dappletSettingsOpened } = useAppSelector(
    state => state.layout,
  )

  const dispatch = useAppDispatch()

  const burgerClickHandler = () => {
    dispatch(setMenuState(!menuOpened))

    if (windowInnerWidth <= 880) dispatch(setDappletSettingsState(false))
  }

  const settingsClickHandler = () => {
    dispatch(setDappletSettingsState(!dappletSettingsOpened))

    if (windowInnerWidth <= 880) dispatch(setMenuState(false))
  }

  return (
    <div className={styles.root}>
      <div className={styles['wrapper']}>
        {windowInnerWidth <= 1300 && (
          <div className={styles['burger']} onClick={burgerClickHandler}>
            <SvgIcon icon={'burger'} />
          </div>
        )}

        {windowInnerWidth <= 880 && (
          <span className={styles['logo-text']}>
            Dapplets Project<span className={styles['red-dot']}>.</span>
          </span>
        )}

        {windowInnerWidth > 880 && <ExtensionState />}
      </div>

      <div onClick={settingsClickHandler}>
        <HeaderSettings />
      </div>
    </div>
  )
}

export default Header
