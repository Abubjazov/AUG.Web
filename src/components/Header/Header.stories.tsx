import type { Meta, StoryObj } from '@storybook/react'
import MockedProvider from 'mockData/mockedReduxProvider'

import Header, { HeaderProps } from './Header'

import '/src/index.css'

const meta: Meta<HeaderProps> = {
  component: Header,
  title: 'COMPONENTS/Header',
  parameters: {
    docs: {
      description: {
        component: 'Interface component - Header',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    windowInnerWidth: {
      description:
        'Window Inner Width. Points of manifestation of effects: 1300px, 880px',
    },
  },
}

export default meta

type Story = StoryObj<HeaderProps>

export const Default: Story = {
  args: { windowInnerWidth: 1920 },
  decorators: [
    Story => (
      <MockedProvider>
        {/* <div style={{ width: '300px' }}> */}
        <Story />
        {/* </div> */}
      </MockedProvider>
    ),
  ],
}
