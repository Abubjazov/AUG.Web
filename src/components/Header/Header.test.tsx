import { screen, fireEvent, render } from '@testing-library/react'
import {
  mockedReduxProvider as Provider,
  mockedStore,
} from 'mockData/mockedReduxProvider'

import Header from './Header'

describe('Header', () => {
  test('should render Header default: windowInnerWidth = 1920px', () => {
    const { asFragment } = render(
      <Provider>
        <Header windowInnerWidth={1920} />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()

    const dsoState = mockedStore.getState().layout.dappletSettingsOpened

    fireEvent.click(screen.getByTestId('header-settings-button'))
    expect(mockedStore.getState().layout.dappletSettingsOpened).toEqual(
      !dsoState,
    )
  })

  test('should render Header default: windowInnerWidth = 1300px', () => {
    const { asFragment } = render(
      <Provider>
        <Header windowInnerWidth={1300} />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()

    const moState = mockedStore.getState().layout.menuOpened

    fireEvent.click(screen.getByTestId('header-burger-button'))
    expect(mockedStore.getState().layout.menuOpened).toEqual(!moState)

    const dsoState = mockedStore.getState().layout.dappletSettingsOpened

    fireEvent.click(screen.getByTestId('header-settings-button'))
    expect(mockedStore.getState().layout.dappletSettingsOpened).toEqual(
      !dsoState,
    )
  })

  test('should render Header default: windowInnerWidth = 880px', () => {
    const { asFragment } = render(
      <Provider>
        <Header windowInnerWidth={880} />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()

    const moState = mockedStore.getState().layout.menuOpened

    fireEvent.click(screen.getByTestId('header-burger-button'))
    expect(mockedStore.getState().layout.menuOpened).toEqual(!moState)

    const dsoState = mockedStore.getState().layout.dappletSettingsOpened

    fireEvent.click(screen.getByTestId('header-settings-button'))
    expect(mockedStore.getState().layout.dappletSettingsOpened).toEqual(
      !dsoState,
    )
  })
})
