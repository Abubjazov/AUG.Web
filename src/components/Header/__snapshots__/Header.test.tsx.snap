// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render Header default: windowInnerWidth = 880px 1`] = `
<DocumentFragment>
  <div
    class="_root_63c2ea"
  >
    <div
      class="_wrapper_63c2ea"
    >
      <div
        class="_burger_63c2ea"
        data-testid="header-burger-button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 18 10"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="0.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="4.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="8.5"
          />
        </svg>
      </div>
      <span
        class="_logo-text_63c2ea"
      >
        Dapplets Project
        <span
          class="_red-dot_63c2ea"
        >
          .
        </span>
      </span>
    </div>
    <div
      data-testid="header-settings-button"
    >
      <div
        class="_root_d8ac16"
      >
        <svg
          fill="none"
          height="19"
          viewBox="0 0 20 19"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.7218 3.875H16.8746C16.5273 2.25781 15.1385 1.0625 13.4718 1.0625C11.8051 1.0625 10.4162 2.25781 10.069 3.875H0.277344V5.28125H10.069C10.4162 6.89844 11.8051 8.09375 13.4718 8.09375C15.1385 8.09375 16.5273 6.89844 16.8746 5.28125H19.7218V3.875ZM13.4718 6.6875C12.2912 6.6875 11.3885 5.77344 11.3885 4.57812C11.3885 3.38281 12.2912 2.46875 13.4718 2.46875C14.6523 2.46875 15.5551 3.38281 15.5551 4.57812C15.5551 5.77344 14.6523 6.6875 13.4718 6.6875Z"
            fill="#787878"
          />
          <path
            d="M0.277344 15.125H3.12457C3.47179 16.7422 4.86068 17.9375 6.52734 17.9375C8.19401 17.9375 9.5829 16.7422 9.93012 15.125H19.7218V13.7188H9.93012C9.5829 12.1016 8.19401 10.9062 6.52734 10.9062C4.86068 10.9062 3.47179 12.1016 3.12457 13.7188H0.277344V15.125ZM6.52734 12.3125C7.7079 12.3125 8.61068 13.2266 8.61068 14.4219C8.61068 15.6172 7.7079 16.5312 6.52734 16.5312C5.34679 16.5312 4.44401 15.6172 4.44401 14.4219C4.44401 13.2266 5.34679 12.3125 6.52734 12.3125Z"
            fill="#787878"
          />
        </svg>
        <span
          class="_text_d8ac16"
        >
          Settings
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header > should render Header default: windowInnerWidth = 1300px 1`] = `
<DocumentFragment>
  <div
    class="_root_63c2ea"
  >
    <div
      class="_wrapper_63c2ea"
    >
      <div
        class="_burger_63c2ea"
        data-testid="header-burger-button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 18 10"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="0.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="4.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="8.5"
          />
        </svg>
      </div>
      <div
        class="_root_8331d9"
      >
        <svg
          fill="none"
          height="17"
          viewBox="0 0 18 17"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_18_300)"
          >
            <path
              d="M15.3552 7.09004V6.59004C15.3646 5.09936 14.7834 3.66561 13.7387 2.60219C12.694 1.53877 11.2708 0.93219 9.78023 0.915039H9.72523C8.43627 0.918753 7.18746 1.36387 6.18674 2.17627C5.18602 2.98868 4.49376 4.11935 4.22523 5.38004C3.23465 5.38004 2.28464 5.77355 1.58419 6.474C0.883742 7.17444 0.490234 8.12446 0.490234 9.11504C0.490234 10.1056 0.883742 11.0556 1.58419 11.7561C2.28464 12.4565 3.23465 12.85 4.22523 12.85H14.5652C15.2788 12.8578 15.9706 12.6048 16.5109 12.1387C17.0512 11.6726 17.4028 11.0253 17.4998 10.3183C17.5967 9.61139 17.4324 8.89334 17.0375 8.29896C16.6427 7.70458 16.0445 7.27472 15.3552 7.09004Z"
              fill="#0085FF"
            />
            <path
              d="M14.79 15.0899H9.42504V13.8899H8.42504V15.0899H3.04004C2.90743 15.0899 2.78025 15.1426 2.68649 15.2363C2.59272 15.3301 2.54004 15.4573 2.54004 15.5899C2.54004 15.7225 2.59272 15.8497 2.68649 15.9434C2.78025 16.0372 2.90743 16.0899 3.04004 16.0899H14.79C14.9226 16.0899 15.0498 16.0372 15.1436 15.9434C15.2374 15.8497 15.29 15.7225 15.29 15.5899C15.29 15.4573 15.2374 15.3301 15.1436 15.2363C15.0498 15.1426 14.9226 15.0899 14.79 15.0899Z"
              fill="#0085FF"
            />
          </g>
          <defs>
            <clippath
              id="clip0_18_300"
            >
              <rect
                fill="white"
                height="16"
                transform="translate(0 0.5)"
                width="18"
              />
            </clippath>
          </defs>
        </svg>
        <span
          class="_text_8331d9"
        >
          Extension state: 
          <span
            class="_state_8331d9"
          >
            Active
          </span>
        </span>
      </div>
    </div>
    <div
      data-testid="header-settings-button"
    >
      <div
        class="_root_d8ac16"
      >
        <svg
          fill="none"
          height="19"
          viewBox="0 0 20 19"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.7218 3.875H16.8746C16.5273 2.25781 15.1385 1.0625 13.4718 1.0625C11.8051 1.0625 10.4162 2.25781 10.069 3.875H0.277344V5.28125H10.069C10.4162 6.89844 11.8051 8.09375 13.4718 8.09375C15.1385 8.09375 16.5273 6.89844 16.8746 5.28125H19.7218V3.875ZM13.4718 6.6875C12.2912 6.6875 11.3885 5.77344 11.3885 4.57812C11.3885 3.38281 12.2912 2.46875 13.4718 2.46875C14.6523 2.46875 15.5551 3.38281 15.5551 4.57812C15.5551 5.77344 14.6523 6.6875 13.4718 6.6875Z"
            fill="#787878"
          />
          <path
            d="M0.277344 15.125H3.12457C3.47179 16.7422 4.86068 17.9375 6.52734 17.9375C8.19401 17.9375 9.5829 16.7422 9.93012 15.125H19.7218V13.7188H9.93012C9.5829 12.1016 8.19401 10.9062 6.52734 10.9062C4.86068 10.9062 3.47179 12.1016 3.12457 13.7188H0.277344V15.125ZM6.52734 12.3125C7.7079 12.3125 8.61068 13.2266 8.61068 14.4219C8.61068 15.6172 7.7079 16.5312 6.52734 16.5312C5.34679 16.5312 4.44401 15.6172 4.44401 14.4219C4.44401 13.2266 5.34679 12.3125 6.52734 12.3125Z"
            fill="#787878"
          />
        </svg>
        <span
          class="_text_d8ac16"
        >
          Settings
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header > should render Header default: windowInnerWidth = 1920px 1`] = `
<DocumentFragment>
  <div
    class="_root_63c2ea"
  >
    <div
      class="_wrapper_63c2ea"
    >
      <div
        class="_root_8331d9"
      >
        <svg
          fill="none"
          height="17"
          viewBox="0 0 18 17"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_18_300)"
          >
            <path
              d="M15.3552 7.09004V6.59004C15.3646 5.09936 14.7834 3.66561 13.7387 2.60219C12.694 1.53877 11.2708 0.93219 9.78023 0.915039H9.72523C8.43627 0.918753 7.18746 1.36387 6.18674 2.17627C5.18602 2.98868 4.49376 4.11935 4.22523 5.38004C3.23465 5.38004 2.28464 5.77355 1.58419 6.474C0.883742 7.17444 0.490234 8.12446 0.490234 9.11504C0.490234 10.1056 0.883742 11.0556 1.58419 11.7561C2.28464 12.4565 3.23465 12.85 4.22523 12.85H14.5652C15.2788 12.8578 15.9706 12.6048 16.5109 12.1387C17.0512 11.6726 17.4028 11.0253 17.4998 10.3183C17.5967 9.61139 17.4324 8.89334 17.0375 8.29896C16.6427 7.70458 16.0445 7.27472 15.3552 7.09004Z"
              fill="#0085FF"
            />
            <path
              d="M14.79 15.0899H9.42504V13.8899H8.42504V15.0899H3.04004C2.90743 15.0899 2.78025 15.1426 2.68649 15.2363C2.59272 15.3301 2.54004 15.4573 2.54004 15.5899C2.54004 15.7225 2.59272 15.8497 2.68649 15.9434C2.78025 16.0372 2.90743 16.0899 3.04004 16.0899H14.79C14.9226 16.0899 15.0498 16.0372 15.1436 15.9434C15.2374 15.8497 15.29 15.7225 15.29 15.5899C15.29 15.4573 15.2374 15.3301 15.1436 15.2363C15.0498 15.1426 14.9226 15.0899 14.79 15.0899Z"
              fill="#0085FF"
            />
          </g>
          <defs>
            <clippath
              id="clip0_18_300"
            >
              <rect
                fill="white"
                height="16"
                transform="translate(0 0.5)"
                width="18"
              />
            </clippath>
          </defs>
        </svg>
        <span
          class="_text_8331d9"
        >
          Extension state: 
          <span
            class="_state_8331d9"
          >
            Active
          </span>
        </span>
      </div>
    </div>
    <div
      data-testid="header-settings-button"
    >
      <div
        class="_root_d8ac16"
      >
        <svg
          fill="none"
          height="19"
          viewBox="0 0 20 19"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.7218 3.875H16.8746C16.5273 2.25781 15.1385 1.0625 13.4718 1.0625C11.8051 1.0625 10.4162 2.25781 10.069 3.875H0.277344V5.28125H10.069C10.4162 6.89844 11.8051 8.09375 13.4718 8.09375C15.1385 8.09375 16.5273 6.89844 16.8746 5.28125H19.7218V3.875ZM13.4718 6.6875C12.2912 6.6875 11.3885 5.77344 11.3885 4.57812C11.3885 3.38281 12.2912 2.46875 13.4718 2.46875C14.6523 2.46875 15.5551 3.38281 15.5551 4.57812C15.5551 5.77344 14.6523 6.6875 13.4718 6.6875Z"
            fill="#787878"
          />
          <path
            d="M0.277344 15.125H3.12457C3.47179 16.7422 4.86068 17.9375 6.52734 17.9375C8.19401 17.9375 9.5829 16.7422 9.93012 15.125H19.7218V13.7188H9.93012C9.5829 12.1016 8.19401 10.9062 6.52734 10.9062C4.86068 10.9062 3.47179 12.1016 3.12457 13.7188H0.277344V15.125ZM6.52734 12.3125C7.7079 12.3125 8.61068 13.2266 8.61068 14.4219C8.61068 15.6172 7.7079 16.5312 6.52734 16.5312C5.34679 16.5312 4.44401 15.6172 4.44401 14.4219C4.44401 13.2266 5.34679 12.3125 6.52734 12.3125Z"
            fill="#787878"
          />
        </svg>
        <span
          class="_text_d8ac16"
        >
          Settings
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
